froi/widgets/treemodel.py

```diff
@@ -71,18 +74,116 @@ def data(self, index, role):
         if not index.isValid():
             return None
 
-        if role != Qt.DisplayRole:
-            return None
-
         item = index.internalPointer()
-        return item.name
+
+        if item in self._data:
+            if role == Qt.UserRole + 2:
+                return item.get_alpha()
+            elif role == Qt.UserRole + 3:
+                return item.get_colormap()
+        else:
+            if role == Qt.UserRole:
+                return item.get_min()
+            elif role == Qt.UserRole + 1:
+                return item.get_max()
+            elif role == Qt.UserRole + 2:
+                return item.get_alpha()
+            elif role == Qt.UserRole + 3:
+                return item.get_colormap()
+
+        if role == Qt.DisplayRole or role == Qt.EditRole:
+           return item.name
+
+        if role == Qt.CheckStateRole:
+            if index.column() == 0:
+                return self.checkState(index)
+
+    def checkState(self, index):
```

Comment:
I'm wondering what the relation between ckeckLisk(I think it may be ckeckList, right?) and the `visible` variable in the data object.